dask_cuda/benchmarks/utils.py

```diff
@@ -0,0 +1,179 @@
+        "The list begins with the host where the scheduler will be launched "
+        "followed by any number of workers, with a minimum of 1 worker. "
+        "Requires --multi-node, ignored otherwise.",
+    )
+
+    for args in args_list:
+        name = args.pop("name")
+        if not isinstance(name, list):
+            name = [name]
+        parser.add_argument(*name, **args)
+
+    parser.set_defaults(
+        enable_tcp_over_ucx=True, enable_infiniband=True, enable_nvlink=True
+    )
+    args = parser.parse_args()
+
+    if args.protocol == "tcp":
+        args.enable_tcp_over_ucx = False
+        args.enable_infiniband = False
+        args.enable_nvlink = False
+
+    if args.multi_node and len(args.hosts.split(",")) < 2:
+        raise ValueError("--multi-node requires at least 2 hosts")
+
+    return args
+
+
+def get_cluster_options(args):
+    if args.multi_node is True:
+        Cluster = SSHCluster
+        cluster_args = [args.hosts.split(",")]
+        scheduler_addr = args.protocol + "://" + cluster_args[0][0] + ":8786"
+
+        worker_options = {}
+
+        # This looks counterintuitive but adding the variable name with
+        # an empty string is how we can enable CLI booleans currently,
+        # note that SSHCluster uses the dask-cuda-worker CLI.
+        if args.enable_tcp_over_ucx:
+            worker_options["enable_tcp_over_ucx"] = ""
+        if args.enable_nvlink:
+            worker_options["enable_nvlink"] = ""
+        if args.enable_infiniband:
+            worker_options["enable_infiniband"] = ""
+
+        if args.ucx_net_devices:
+            worker_options["ucx_net_devices"] = args.ucx_net_devices
+
+        cluster_kwargs = {
+            "connect_options": {"known_hosts": None},
+            "scheduler_options": {"protocol": args.protocol},
+            "worker_module": "dask_cuda.dask_cuda_worker",
+            "worker_options": worker_options,
+            # "n_workers": len(args.devs.split(",")),
+            # "CUDA_VISIBLE_DEVICES": args.devs,
+        }
+    else:
+        Cluster = LocalCUDACluster
+        scheduler_addr = None
+        cluster_args = []
+        cluster_kwargs = {
+            "protocol": args.protocol,
+            "n_workers": len(args.devs.split(",")),
+            "CUDA_VISIBLE_DEVICES": args.devs,
+            "ucx_net_devices": args.ucx_net_devices,
+            "enable_tcp_over_ucx": args.enable_tcp_over_ucx,
+            "enable_infiniband": args.enable_infiniband,
+            "enable_nvlink": args.enable_nvlink,
+        }
+
+    return {
+        "class": Cluster,
+        "args": cluster_args,
+        "kwargs": cluster_kwargs,
+        "scheduler_addr": scheduler_addr,
+    }
+
+
+def get_scheduler_workers(dask_scheduler=None):
+    return dask_scheduler.workers
+
+
+def setup_memory_pool(pool_size=None, disable_pool=False):
+    import rmm
+    import cupy
+
+    rmm.reinitialize(
```

Comment:
If the the `disable_pool` is True should we still set the cupy allocator ?

Reply:
TBH I just moved this function, I didn't change anything. I think it's fine (perhaps even preferred) to use the same allocator, even if we don't enable the pool it's fine to use RMM as the allocator for CuPy, that shouldn't cause problems.